### PR TITLE
refactor: replace char_u with char

### PR DESCRIPTION
**Pull request by dundargoc**
Work on https://github.com/neovim/neovim/issues/459

### CODE REVIEW

File: src/nvim/tag.c [hunk 1]
```diff
@@ -1411,7 +1411,7 @@ int find_tags(char_u *pat, int *num_matches, char ***matchesp, int flags, int mi
   int matchoff = 0;
   int save_emsg_off;
 
-  char_u *mfp;
+  char *mfp;
```
Comment (zeertzjq) [inline src/nvim/tag.c:1414]: This looks strange as there are a lot of `sizeof(char_u)` and `sizeof(char_u *)` below.

Comment (dundargoc) [reply src/nvim/tag.c:1414]: Aren't they the same size anyway?

I can change the sizeof types.

**Comment by clason** [comment]
needs a rebase now...

**Comment by zeertzjq** [comment]
Rebased.

**Comment by bfredl** [comment]
looks fine, anything missing to mark this as ready?

**Comment by dundargoc** [comment]
> looks fine, anything missing to mark this as ready?

asan test is failing.

**Comment by clason** [comment]
asan test is failing everywhere; that hasn't stopped any of the other PRs ;)

**Comment by dundargoc** [comment]
True. Undrafting it.

**Comment by bfredl** [comment]
ASAN is passing again :sparkles:

**Pull request merged**